File: app/src/main/java/com/autoscout24/carfinder/ui/utils/StringUtils.java
package com.autoscout24.carfinder.ui.utils;

import java.text.DecimalFormat;

public class StringUtils {

    public static String isNullorEmpty(String value) {
        if(value != null && !value.isEmpty()) {
            return value;
        }else{
            return "N/A";
        }
    }

    public static String getAttributtedCurrency(double currency){
        DecimalFormat formatter = new DecimalFormat("##,###.00 -");
        return currency > 0 ?  "€ "+formatter.format(currency) : "N/A";
    }

    public static String getAttributedMileage(double mileage){
        DecimalFormat formatter = new DecimalFormat("##,### KM");
        return formatter.format(mileage);
    }
}

File: app/src/main/java/com/autoscout24/carfinder/ui/utils/StringUtils.kt
package com.autoscout24.carfinder.ui.utils

import java.text.DecimalFormat

object StringUtils {
    @JvmStatic
    fun isNullorEmpty(value: String): String {
        var result = ""

        when((value != null && value.isNotBlank())){
            true -> result = value
            false-> result = "N/A"
        }
        return result
    }
    @JvmStatic
    fun getAttributtedCurrency(currency : Double) : String {
        val formatter = DecimalFormat("##,###.00 -")
        return if(currency > 0)  "€ "+formatter.format(currency) else "N/A"
    }
    @JvmStatic
    fun getAttributedMileage(mileage : Double) : String{
        val formatter = DecimalFormat("##,### KM")
        return formatter.format(mileage)
    }

}

